system/systemd.py

```diff
@@ -0,0 +1,329 @@
+            "MountFlags": "0",
+            "Names": "crond.service",
+            "NeedDaemonReload": "no",
+            "Nice": "0",
+            "NoNewPrivileges": "no",
+            "NonBlocking": "no",
+            "NotifyAccess": "none",
+            "OOMScoreAdjust": "0",
+            "OnFailureIsolate": "no",
+            "PermissionsStartOnly": "no",
+            "PrivateNetwork": "no",
+            "PrivateTmp": "no",
+            "RefuseManualStart": "no",
+            "RefuseManualStop": "no",
+            "RemainAfterExit": "no",
+            "Requires": "basic.target",
+            "Restart": "no",
+            "RestartUSec": "100ms",
+            "Result": "success",
+            "RootDirectoryStartOnly": "no",
+            "SameProcessGroup": "no",
+            "SecureBits": "0",
+            "SendSIGHUP": "no",
+            "SendSIGKILL": "yes",
+            "Slice": "system.slice",
+            "StandardError": "inherit",
+            "StandardInput": "null",
+            "StandardOutput": "journal",
+            "StartLimitAction": "none",
+            "StartLimitBurst": "5",
+            "StartLimitInterval": "10000000",
+            "StatusErrno": "0",
+            "StopWhenUnneeded": "no",
+            "SubState": "running",
+            "SyslogLevelPrefix": "yes",
+            "SyslogPriority": "30",
+            "TTYReset": "no",
+            "TTYVHangup": "no",
+            "TTYVTDisallocate": "no",
+            "TimeoutStartUSec": "1min 30s",
+            "TimeoutStopUSec": "1min 30s",
+            "TimerSlackNSec": "50000",
+            "Transient": "no",
+            "Type": "simple",
+            "UMask": "0022",
+            "UnitFileState": "enabled",
+            "WantedBy": "multi-user.target",
+            "Wants": "system.slice",
+            "WatchdogTimestampMonotonic": "0",
+            "WatchdogUSec": "0",
+        }
+'''
+
+import os
+import glob
+from ansible.module_utils.basic import AnsibleModule
+
+# ===========================================
+# Main control flow
+
+if __name__ == '__main__':
+    # init
+    module = AnsibleModule(
+        argument_spec = dict(
+                name = dict(required=True, type='str'),
+                state = dict(choices=[ 'started', 'stopped', 'restarted', 'reloaded'], type='str'),
+                enabled = dict(type='bool'),
+                masked = dict(type='bool'),
+                daemon_reload= dict(type='bool', default=False, aliases=['daemon-reload']),
+            ),
+            supports_check_mode=True,
+            required_one_of=[['state', 'enabled', 'masked', 'daemon_reload']],
+        )
+
+    # initialize
+    systemctl = module.get_bin_path('systemctl')
+    unit = module.params['name']
+    rc = 0
+    out = err = ''
+    result = {
+        'name':  unit,
+        'changed': False,
+        'status': {},
+    }
+
+    #TODO: check if service exists
+    (rc, out, err) = module.run_command("%s show '%s'" % (systemctl, unit))
```

Comment:
If we are running the cli tools, it may be useful to add a systemd aware 'module.run_command' analog. 

- could be useful to be able to grab the journal entries related to running of the command/unit/pid to potentially include in the return code or debugging info

- single place to specify host/machine/user for --user/--host/--machine options